pkg/engine/planner/logical/node_parse.go

```diff
@@ -0,0 +1,62 @@
+package logical
+
+import (
+	"fmt"
+
+	"github.com/grafana/loki/v3/pkg/engine/planner/schema"
+)
+
+// ParserKind represents the type of parser to use
+type ParserKind int
+
+const (
+	ParserLogfmt ParserKind = iota
+	ParserJSON
+)
+
+// NumericType represents the numeric type to cast a parsed field to
+type NumericType int
+
+const (
+	NumericNone NumericType = iota
+	NumericInt64
+	NumericFloat64
+)
+
+// Parse represents a parsing instruction that extracts fields from log lines.
+// It takes a table relation as input and produces a new table relation with
+// additional columns for the parsed fields.
+type Parse struct {
+	id string
+
+	Table         Value // The table relation to parse from
+	Kind          ParserKind
+	RequestedKeys []string
+	NumericHints  map[string]NumericType // Fields that should be cast to numeric types
```

Comment:
Both of these fields are closer to the domain of the physical planner and optimization passes where we can push down the requested keys / hints. 

I think we should take these out of the logical planner entirely.